dedupe/notebooks/lac_ecr_eda.ipynb

```diff
@@ -0,0 +1,768 @@
+    "class eCR:\n",
+    "    def __init__(self, path):\n",
+    "        self.path = path\n",
+    "        self.tree = etree.parse(path)\n",
+    "        self.root = self.tree.getroot()\n",
+    "        self.ns = self.root.nsmap\n",
+    "        self.ns['default'] = self.ns.pop(None)\n",
+    "\n",
+    "class RR(eCR):\n",
+    "    def __init__(self, path):\n",
+    "        super().__init__(path)\n",
+    "\n",
+    "class eICR(eCR):\n",
+    "    def __init__(self, path):\n",
+    "        super().__init__(path)\n",
+    "    \n",
+    "    def get_patient_id(self):\n",
+    "        return self.root.find('.//default:patient/default:id', self.ns).attrib['extension']\n",
+    "    \n",
+    "    def get_patient_name(self):\n",
+    "        given = self.root.find('.//default:patient/default:name/default:given', self.ns).text\n",
+    "        family = self.root.find('.//default:patient/default:name/default:family', self.ns).text\n",
+    "        return f\"{given} {family}\"\n",
+    "    \n",
+    "    def get_patient_dob(self):\n",
+    "        return self.root.find('.//default:patient/default:birthTime', self.ns).attrib['value']\n",
+    "    \n",
+    "    def get_send_date(self):\n",
+    "        time = self.root.find('.//default:effectiveTime', self.ns)\n",
+    "        if time is not None:\n",
+    "            return self._format_date_time(time.attrib.get('value'))\n",
+    "        \n",
+    "    def get_encounter_id(self):\n",
+    "        id = self.root.find('.//default:encounter/default:id', self.ns)\n",
+    "        if id is not None:\n",
+    "            return id.attrib.get('extension')\n",
+    "\n",
+    "    def get_set_id(self):\n",
+    "        id = self.root.find('.//default:setId', self.ns)\n",
+    "        if id is not None:\n",
+    "            return id.attrib.get('extension')\n",
+    "\n",
+    "    def get_ecr_version(self):\n",
+    "        version = self.root.find('.//default:versionNumber', self.ns)\n",
+    "        if version is not None:\n",
+    "            return version.attrib.get('value') or version.text\n",
+    "\n",
+    "    def get_sections(self):\n",
+    "        return self.root.findall('.//default:section', self.ns)\n",
+    "    \n",
+    "    def _format_date_time(self, date_time):\n",
+    "        if date_time is not None:\n",
+    "            return datetime.strptime(date_time, '%Y%m%d%H%M%S%z').isoformat()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "section_name_counts = {}\n",
+    "\n",
+    "for dir in all_dirs:\n",
+    "    # read in the CDA_eICR.xml file, get the sections, and print the section titles\n",
+    "    file_path = Path(\"./../data/LAC_DATA/\" + dir + \"/CDA_eICR.xml\")\n",
+    "    ecr = eICR(file_path)\n",
+    "    sections = ecr.get_sections()\n",
+    "    for section in sections:\n",
+    "        section_name = section.find('.//default:title', ecr.ns).text\n",
```

Comment:
I wonder if LOINC/SNOWMED codes are more standard than section titles across eCRs. I'll look into this!

Reply:
I believe this is definitely the case and is likely the better way to extract sections. Adding that analysis to this notebook would be very beneficial!